container-engine-lib/lib/backend_impls/docker/object_attributes_provider/enclave_object_attributes_provider.go

```diff
@@ -321,6 +331,56 @@ func (provider *dockerEnclaveObjectAttributesProviderImpl) ForSingleFilesArtifac
 	return objectAttributes, nil
 }
 
+// In Docker we get one volume per persistent directory
+// TODO: refactor with ForSingleFilesArtifactExpansionVolume maybe
+func (provider *dockerEnclaveObjectAttributesProviderImpl) ForSinglePersistentDirectoryVolume(
+	serviceUUID service.ServiceUUID,
+	persistentKey service_directory.DirectoryPersistentKey,
+) (
+	DockerObjectAttributes,
+	error,
+) {
+	serviceUuidStr := string(serviceUUID)
+
+	guidStr, err := uuid_generator.GenerateUUIDString()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred generating a UUID for the persistent directory volume for service '%v'", serviceUuidStr)
+	}
+
+	hasher := md5.New()
+	hasher.Write([]byte(serviceUUID))
+	hasher.Write([]byte(persistentKey))
+	persistentKeyHash := hex.EncodeToString(hasher.Sum(nil))
+
+	name, err := provider.getNameForEnclaveObject([]string{
+		persistentServiceDirectoryNameFragment,
+		persistentKeyHash,
+	})
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating the files artifact expansion volume name object using GUID '%v' and service GUID '%v'", guidStr, serviceUuidStr)
+	}
+
+	labels, err := provider.getLabelsForEnclaveObjectWithGUID(guidStr)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred getting labels for files artifact expansion volume with UUID '%v'", guidStr)
+	}
+
+	serviceUuidLabelValue, err := docker_label_value.CreateNewDockerLabelValue(serviceUuidStr)
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred creating a Docker label value from service GUID string '%v'", serviceUuidStr)
+	}
+	labels[label_key_consts.UserServiceGUIDDockerLabelKey] = serviceUuidLabelValue
+	labels[label_key_consts.VolumeTypeDockerLabelKey] = label_value_consts.PersistentDirectoryVolumeTypeDockerLabelValue
+	// TODO Create a KurtosisResourceDockerLabelKey object, like Kubernetes, and apply the "user-service" label here?
```

Comment:
is this TODO for later? or is this something missing on this PR?

Reply:
I can image that it's for later but I'll ask just to double check

Reply:
I have to admit this is from a copy paste from `ForSingleFilesArtifactExpansionVolume` and I am not sure we want to do that. I'll leave it around for now. I think until we should not add this label unless we have a concrete use case for it. If you can think of one, I'll add :)